server/src/main/java/org/elasticsearch/index/IndexSortConfig.java

```diff
@@ -152,10 +153,16 @@ public IndexSortConfig(IndexSettings indexSettings) {
         }
 
         List<String> fields = INDEX_SORT_FIELD_SETTING.get(settings);
-        this.sortSpecs = fields.stream().map((name) -> new FieldSortSpec(name)).toArray(FieldSortSpec[]::new);
+        if (this.indexMode == IndexMode.LOGS && fields.isEmpty()) {
+            fields = List.of("hostname", DataStream.TIMESTAMP_FIELD_NAME);
```

Comment:
This should be `host.name`. See also https://www.elastic.co/guide/en/ecs/current/ecs-host.html#field-host-name.